src/gpu/intel/jit/ngen/ngen_level_zero.hpp

```diff
@@ -21,13 +21,100 @@
 
 #include "gpu/intel/sycl/l0/level_zero/ze_api.h"
 
+#if defined(__linux__)
+#include <dlfcn.h>
+#elif defined(_WIN32)
+#include "windows.h"
+#else
+#error "Level Zero is supported on Linux and Windows only"
+#endif
+
 #include <sstream>
 
 #include "ngen_elf.hpp"
 #include "ngen_interface.hpp"
 
 namespace NGEN_NAMESPACE {
 
+// Dynamically loaded level_zero functions
+namespace {
+
+void *find_ze_symbol(const char *symbol) {
+#if defined(__linux__)
+    void *handle = dlopen("libze_loader.so.1", RTLD_NOW | RTLD_LOCAL);
+#elif defined(_WIN32)
+    // Use LOAD_LIBRARY_SEARCH_SYSTEM32 flag to avoid DLL hijacking issue.
+    HMODULE handle = LoadLibraryExA(
+            "ze_loader.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
+#endif
+    if (!handle) {
+        assert(!"not expected");
```

Comment:
nGEN uses named exception classes (like `level_zero_error` below) instead of assertions to handle these kinds of situations, so it would be nice to match that in this code.

Reply:
Done. Thanks!